### PR TITLE
add Start() method in Filter interface

### PR DESCRIPTION
**Pull request by tyroneyu**
## Description
Please include a summary of the change, motivation and context.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: sdk/filter/multifilter.go [hunk 1]
```diff
@@ -14,6 +14,8 @@ func NewMultiFilter(filter ...Filter) *MultiFilter {
 	return &MultiFilter{filters: filter}
 }
 
+func (m *MultiFilter) Start() {}
```
Comment (ryanericson) [inline sdk/filter/multifilter.go:17]: `MultiFilter.Start` Should call Start for each filter.

**Review by ryanericson — APPROVED** [review]

File: sdk/filter/filter.go [hunk 2]
```diff
@@ -4,6 +4,9 @@ import "github.com/hypertrace/goagent/sdk"
 
 // Filter evaluates whether request should be blocked, `true` blocks the request and `false` continues it.
 type Filter interface {
+	// Start the filter to allow evaluating requests
+	Start()
```
Comment (ryanericson) [inline sdk/filter/filter.go:8]: Would be good to change example in README.md to include call to `Start`.

File: sdk/filter/multifilter.go (hunk 1 above)
Comment (tyroneyu) [reply sdk/filter/multifilter.go:17]: done

**Review by jcchavezs — CHANGES_REQUESTED** [review]
I am a bit lost on this feature. To get started I'd like to get some questions addressed:
- Why do we need this method?
- Who is calling the Start?
- When/Where is Start called?
- Can start fails? What would be the behaviour if that happens?

I think if we can answers this questions in code, that would really help the design on this feature.

**Comment by tyroneyu** [comment]
Closing cause this interface is not truly needed.

**Pull request closed without merging**